### PR TITLE
Pip package

### PR DESCRIPTION
**Pull request by michael-erasmus**
Hi @davidgasquez 

As we discussed earlier, I would love to share the PR I made for moving the project to a installable package. As I mentioned before, I can still see extracting some of the more generic Redshift data loading code into a separate package as well, but I haven't done that here yet :D

No rush on this one, but I would really appreciate your thoughts/ideas!

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -6,15 +6,7 @@ verify_ssl = true
...
-stripe = "*"
-sqlalchemy = "*"
-"psycopg2" = "*"
-python-dateutil = "*"
-"boto3" = "*"
-smart-open = "*"
-decorator = "*"
-click = "*"
-retrying = "*"
+"e1839a8" = {path = ".", editable = true}
```
Comment (michael-erasmus) [inline Pipfile:9]: Moving all the requirements to the `setup.py` file, pipenv will need to be aware of the package itself.

File: bin/stripe-pipeline [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env python
+
+from stripe_pipeline.cli import cli
```
Comment (michael-erasmus) [inline bin/stripe-pipeline]: A tiny wrapper executable for the `cli` module

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+          'stripe',
+          'sqlalchemy',
+          'psycopg2',
+          'python-dateutil',
+          'boto3',
+          'smart-open',
+          'click',
+          'retrying'
+      ],
+      scripts=['bin/stripe-pipeline'])
```
Comment (michael-erasmus) [inline setup.py]: This refers to the `stripe-pipeline` executable wrapper

File: stripe_pipeline/cli.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import click
+from stripe_pipeline.crawler import crawler
+
+
+@click.group()
+def cli():
+    pass
+
+
+cli.add_command(crawler)
```
Comment (michael-erasmus) [inline stripe_pipeline/cli.py:12]: Write now the cli just has a single `crawler` command, but I plan on adding more commands, for things like initializing the database and possible add-hoc tasks

File: bin/stripe-pipeline (hunk 2 above)
Comment (davidgasquez) [reply bin/stripe-pipeline]: I think this file could be deleted if the setup script uses the [`entrypoint` parameter](http://click.pocoo.org/5/setuptools/#introduction). The entrypoint specifies the name of the binary package and which function should it reference. 

In this case could be something like:

```python
entry_points={
    'console_scripts': [
        'stripe-pipeline=stripe_pipeline.cli.cli'
    ]
}
```

**Review by davidgasquez — APPROVED** [review]
Hey there @michael-erasmus, nice refactoring/packaging! 😄 

I've left one comment but LGTM overall!

File: bin/stripe-pipeline (hunk 2 above)
Comment (michael-erasmus) [reply bin/stripe-pipeline]: Whoa, didn't know about that! I'll try that out!

File: stripe_pipeline/cli.py (hunk 4 above)
Comment (michael-erasmus) [reply stripe_pipeline/cli.py:12]: Thanks @davidgasquez! That worked great, added it in [this commit](https://github.com/bufferapp/stripe-pipeline/pull/1/commits/c97db2db982acf4a64169b45b4903aa64db68323)

**Pull request merged**